File: scripts/generate-icons.js
const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

const logoPath = path.join(__dirname, '../src/assets/tveco-logo.png');
const resDir = path.join(__dirname, '../android/app/src/main/res');

const sizes = [
  { dir: 'mipmap-mdpi', size: 48 },
  { dir: 'mipmap-hdpi', size: 72 },
  { dir: 'mipmap-xhdpi', size: 96 },
  { dir: 'mipmap-xxhdpi', size: 144 },
  { dir: 'mipmap-xxxhdpi', size: 192 },
];

async function generateIcons() {
  console.log('Generating Android icons from TVECO logo...');
  
  for (const { dir, size } of sizes) {
    const dirPath = path.join(resDir, dir);
    
    // Ensure directory exists
    if (!fs.existsSync(dirPath)) {
      fs.mkdirSync(dirPath, { recursive: true });
      console.log(`✓ Created ${dir}`);
    }

    // Generate ic_launcher.png
    await sharp(logoPath)
      .resize(size, size, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
      .png()
      .toFile(path.join(dirPath, 'ic_launcher.png'));
    
    // Generate ic_launcher_foreground.png (for adaptive icons)
    await sharp(logoPath)
      .resize(size, size, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
      .png()
      .toFile(path.join(dirPath, 'ic_launcher_foreground.png'));
    
    // Generate ic_launcher_round.png
    await sharp(logoPath)
      .resize(size, size, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
      .png()
      .toFile(path.join(dirPath, 'ic_launcher_round.png'));

    console.log(`✓ Generated icons for ${dir} (${size}x${size}px)`);
  }

  console.log('✓ All Android icons generated successfully!');
}

generateIcons().catch(err => {
  console.error('Error generating icons:', err);
  process.exit(1);
});

File: scripts/generate-icons.py
#!/usr/bin/env python3
from PIL import Image
import os
import sys

logo_path = '../src/assets/tveco-logo.png'
res_dir = '../android/app/src/main/res'

sizes = [
    ('mipmap-mdpi', 48),
    ('mipmap-hdpi', 72),
    ('mipmap-xdpi', 96),
    ('mipmap-xxhdpi', 144),
    ('mipmap-xxxhdpi', 192),
]

def generate_icons():
    print('Generating Android icons from TVECO logo...')
    
    # Open the original logo
    try:
        img = Image.open(logo_path)
        print(f'✓ Loaded logo: {logo_path} ({img.size[0]}x{img.size[1]}px)')
    except Exception as e:
        print(f'✗ Error loading logo: {e}')
        sys.exit(1)
    
    for dir_name, size in sizes:
        dir_path = os.path.join(res_dir, dir_name)
        
        # Create directory if it doesn't exist
        os.makedirs(dir_path, exist_ok=True)
        
        # Resize image
        resized = img.resize((size, size), Image.Resampling.LANCZOS)
        
        # Save icon files
        for icon_name in ['ic_launcher.png', 'ic_launcher_foreground.png', 'ic_launcher_round.png']:
            icon_path = os.path.join(dir_path, icon_name)
            resized.save(icon_path, 'PNG')
        
        print(f'✓ Generated icons for {dir_name} ({size}x{size}px)')
    
    print('✓ All Android icons generated successfully!')

if __name__ == '__main__':
    generate_icons()
